packages/peregrine/lib/talons/CheckoutPage/useShippingMethod.js

```diff
@@ -156,62 +95,81 @@ export const useShippingMethod = props => {
-                    .selected_shipping_method;
-
-            const nextDisplayState = chosenShippingMethod
-                ? displayStates.DONE
-                : displayStates.EDITING;
-
-            setDisplayState(nextDisplayState);
-        } catch {
-            setDisplayState(displayStates.EDITING);
+        if (!data) return;
+
+        // Sorts available shipping methods by price.
+        const byPrice = (a, b) => a.amount.value - b.amount.value;
+
+        // Adds a serialized property to available shipping method objects
+        // so they can be selected in the radio group.
+        const addSerializedProperty = shippingMethod => {
+            const serializedValue = serializeShippingMethod(shippingMethod);
+
+            return {
+                ...shippingMethod,
+                serializedValue
+            };
+        };
+
+        // Determine the "primary" shipping address by using
+        // the first shipping address on the cart.
+        const primaryShippingAddress = data.cart.shipping_addresses[0];
+
+        // Shape the list of available shipping methods.
+        // Sort them by price and add a serialized property to each.
+        const rawShippingMethods =
+            primaryShippingAddress.available_shipping_methods;
+        const shippingMethodsByPrice = [...rawShippingMethods].sort(byPrice);
+        const shippingMethods = shippingMethodsByPrice.map(
+            addSerializedProperty
+        );
+        setShippingMethods(shippingMethods);
+
+        // Determine the selected shipping method.
+        // If we don't have one, default to the lowest cost shipping method instead.
+        const selectedMethod = primaryShippingAddress.selected_shipping_method;
+        setSelectedShippingMethod(selectedMethod);
+
+        // If we have data  we should tell the checkout page
+        // so that we set the next step correctly.
+        if (selectedMethod) {
+            onSave();
         }
-    }, [chosenShippingMethodData]);
-
-    // Simple heuristic to check shipping data existed prior to this render.
-    // On first submission, when we have data, we should tell the checkout page
-    // so that we set the next step correctly.
-    const hasData =
-        chosenShippingMethodData &&
-        chosenShippingMethodData.cart.shipping_addresses.length &&
-        chosenShippingMethodData.cart.shipping_addresses[0]
-            .selected_shipping_method;
 
-    useEffect(() => {
-        if (hasData) {
-            onSave();
+        // Determine the component's display state.
+        const nextDisplayState = selectedMethod
+            ? displayStates.DONE
+            : displayStates.EDITING;
+        setDisplayState(nextDisplayState);
+
+        // Sometimes the update form will mount before we have a selected
+        // shipping method to put in its initialValues.
+        // When that happens, use the form's api to update the field value.
+        if (updateFormApi && selectedMethod) {
```

Comment:
This opens the door to blowing away a user choice that occurred after the form mounted but before this code executes.

Reply:
Resolved in https://github.com/magento/pwa-studio/pull/2382/commits/ce7eb8c46ffbaeb490a89011e96621076d3ce312.

This was an attempt to keep two different forms in sync. Instead of trying to do that (which leads to the comment above), we now only mount the `UpdateModal` in completed view so that it has a proper `initialValue`.